docs/source/cc_service.md

```diff
@@ -0,0 +1,297 @@
+* **run** - Not required when using chaincode as an external service if the runtime of chaincode running on the external server is manged externally and the chaincode need not be started from this script.
+
+The scripts are required to exist in the peer `/bin` directory:
+```
+    <fully qualified path on the peer's env>
+    └── bin
+        ├── build
+        ├── detect
+        ├── release
+        └── run
+```
+
+### Modify the peer core.yaml to include the externalBuilder
+
+Finally, in order for the peer to use the external builder and launcher scripts, you need to modify the `chaincode` stanza of the peer `core.yaml` file to include the `externalBuilders` configuration element.
+
+```yaml
+externalBuilders:
+     - name: myexternal
+       path: <fully qualified path on the peer's env>   
+```
+
+## External builder and launcher sample scripts
+
+To help understand what each script needs to contain to work with the chaincode as an external service, this section contains samples of  `bin/detect` `bin/build`, `/bin/release`, and `bin/run` scripts.
+
+**Note:** These samples use the `jq` command to parse json. You can run `jq --version` to check if you have it installed. Otherwise, install `jq` or suitably modify the scripts yourself.
+
+### /bin/detect
+
+The `bin/detect script` is responsible for determining whether or not a buildpack should be used to build a chaincode package and launch it. For chaincode as an external service, the script should detect that metadata.json `type` is set to `external`.  The peer invokes detect with two arguments:
+
+```
+bin/detect CHAINCODE_SOURCE_DIR CHAINCODE_METADATA_DIR
+```
+
+A typical `detect` script could contain:
+
+```sh
+
+#!/bin/bash
+
+set -euo pipefail
+
+if [ "$#" -ne 2 ]; then
+    echo "Expected 2 directories got $#"
+    exit 2
+fi
+
+#check if the "type" field is set to "external"
+if [ "$(jq -r .type "$2/metadata.json")" == "external" ]; then
+    exit 0
+fi
+
+exit 1
+```
+Recall that the metadata.json file should contain the following keys:
+
+```json
+{"path":"","type":"external","label":"mycc"}
+```
+
+### /bin/build
+
+The `bin/build` script is responsible for building, compiling, or transforming the contents of a chaincode package into artifacts that can be used by release and run. For chaincode as an external service, the build script copies the `connection.json` to the `BUILD_OUTPUT_DIR`. The peer invokes the build script with three arguments:
+
+```
+bin/build CHAINCODE_SOURCE_DIR CHAINCODE_METADATA_DIR BUILD_OUTPUT_DIR
+```
+
+A typical `build` script could contain:
+
+```sh
+
+#!/bin/bash
+
+set -euo pipefail
+
+if [ "$#" -ne 3 ]; then
+    echo "Expected 3 directories got $#"
+    exit 1
+fi
+
+SOURCE=$1
+OUTPUT=$3
+
+#external chaincodes expect connection.json file in the chaincode package
+if [ ! -f "$SOURCE/connection.json" ]  ; then
+    echo "$SOURCE/connection.json not found" > /tmp/err.txt
```

Comment:
Let's instead follow the pattern:

```
>&2 echo "$SOURCE/connection.json not found"
```

This ensures that the output goes into the peer logs and there's no contention for the tmp file.

Reply:
done